src/syrupy/assertion.py

```diff
@@ -206,7 +206,9 @@ def get_assert_diff(self) -> List[str]:
             # Rotate to place exception with message at first line
             return lines[-1:] + lines[:-1]
         snapshot_data = assertion_result.recalled_data
-        serialized_data = assertion_result.asserted_data or ""
+        if assertion_result.asserted_data is None:
+            assertion_result.asserted_data = ""
```

Comment:
We should avoid mutating the assertion result and instead should conditionally set serialized_data, like so:

```
serialized_data = assertion_result.asserted_data if assertion_result.asserted_data is not None else ""
```